protocol/transport/envelopes_monitor.go

```diff
@@ -172,7 +172,6 @@ func (m *EnvelopesMonitor) handleEnvelopeEvents() {
 	events := make(chan types.EnvelopeEvent, 100) // must be buffered to prevent blocking waku
 	sub := m.w.SubscribeEnvelopeEvents(events)
 	defer func() {
-		close(events)
```

Comment:
Interesting, but who's closing this channel? I don't see geth is closing it on unsubscribe.

https://github.com/status-im/status-go/blob/2f539d3bd23694f1d0db7fdbf1ffde81f42e8d7d/vendor/github.com/ethereum/go-ethereum/event/subscription.go#L38-L44